trace.go

```diff
@@ -138,12 +138,50 @@ func GetFields(err error) map[string]interface{} {
 	if err == nil {
 		return map[string]interface{}{}
 	}
+
+	// for proxyError combine top-level and nested fields.
+	if proxy, ok := err.(proxyError); ok {
+		fields := map[string]interface{}{}
+
+		// nested
+		for field, value := range GetFields(proxy.Err) {
+			fields[field] = value
+		}
+
+		// top-level
+		for field, value := range proxy.GetFields() {
+			fields[field] = value
+		}
+
+		return fields
+	}
+
 	if wrap, ok := err.(Error); ok {
 		return wrap.GetFields()
 	}
+
 	return map[string]interface{}{}
 }
 
+// UnmarshalField attempts to find a given named field.
+// If found, the field is reinterpreted through json.Unmarshal call and stored in unmarshalPtr.
+// Returns true on success and false on failure.
+func UnmarshalField(err error, fieldName string, unmarshalPtr interface{}) bool {
+	fields := GetFields(err)
+
+	fieldValue, ok := fields[fieldName]
+	if !ok {
+		return false
+	}
+
+	out, marshalErr := json.Marshal(fieldValue)
```

Comment:
why not just `err` ? The locality of this error is reply narrow.

Reply:
`err` is a function argument that shouldn't be overwritten, so we need a new name... or rename the functional argument to something else, such as `originalErr`. I'll to the latter as your comment shows the current code may be confusing.